goalc/compiler/compilation/Asm.cpp

```diff
@@ -399,7 +592,37 @@ Val* Compiler::compile_asm_vf_math3(const goos::Object& form,
         src2->print());
   }
 
-  env->emit_ir<IR_VFMath3Asm>(color, dest, src1, src2, kind);
+  int64_t mask;
+  if (!try_getting_constant_integer(args.unnamed.at(3), &mask, env)) {
+    throw_compiler_error(
+        form,
+        "The value {} is invalid for a destination mask, it could not be evaluated as a "
+        "constant integer.",
+        args.unnamed.at(3).print());
+  }
+
+  if (mask < 0 || mask > 15) {
+    throw_compiler_error(
+        form, "The value {} is out of range for a destination mask (0-15 inclusive).", mask);
+  }
+
+  auto temp_reg = env->make_vfr(dest->type());
+
+  // If there is a broadcast register, splat that float across the entire src2 register before
+  // performing the operation For example vaddx.xyzw vf10, vf20, vf30
+  // vf10[x] = vf20[x] + vf30[x]
+  // vf10[y] = vf20[y] + vf30[x]
+  // vf10[z] = vf20[z] + vf30[x]
+  // vf10[w] = vf20[w] + vf30[x]
+  if (broadcastElement != emitter::Register::XMM_ELEMENT::NONE) {
+    env->emit_ir<IR_SplatVF>(color, src2, src2, broadcastElement);
+  }
+
+  // Perform the arithmetic operation on the two vectors into a temporary register
+  env->emit_ir<IR_VFMath3Asm>(color, temp_reg, src1, src2, mask, kind);
```

Comment:
I think we can add a check to see if `mask == 0b1111` and if so optimize this to a single `IR_VFMath3Asm` without the blend at the end.